extension/lib/tracker.js

```diff
@@ -37,87 +31,28 @@ class Tracker {
     reset() {
         // Int. Pointer to an element in readableDomEls. The current container the tracker is in.
         // If there is no tracker yet, this value is null.
-        // Otherwise, it will be in [0, readableDomEls.length)
-        this.containerId = null;
-        // jQuery element. Must be kept in sync with containerId.
-        // This is a performance optimization to address the fact that jQuery lookups are not cached.
-        this.container = null;
-        // Start of currently tracked sentence, relative to container; 0 <= start < len(container.text())
-        this.start = null;
-        // End of currently tracked sentence, relative to container; 0 <= end < len(container.text())
-        this.end = null; 
+        // Otherwise, it will be in [0, doc.getNumSentences())
+        this.sentenceId = null;
     }
 
     /*
     Return: Boolean. Whether or not the tracker is pointing to anything.
     */
     isTracking() {
-        return this.containerId !== null;
+        return this.sentenceId !== null;
     }
 
-    // Returns: List of readable jquery dom elements on web page
-    getReadableDomEls() {
-        return this.readableDomEls;
-    }
-    // Returns: Container ID of current container
-    getContainerId() {
-        return this.containerId;
+    getSentenceId() {
+        return this.sentenceId;
     }
 
     /*
     Point to the first sentence within the given container.
     */
     pointToContainer(containerId) {
-        this.container = this.getContainer(containerId);
```

Comment:
A lot of deletion in this file. The nlp logic now resides in doc.js, which will then be shared by other files.